receiver/prometheusreceiver/internal/otlp_metricfamily.go

```diff
@@ -254,3 +262,58 @@ func (mf *metricFamilyPdata) Add(metricName string, ls labels.Labels, t int64, v
 
 	return nil
 }
+
+// getGroups to return groups in insertion order
+func (mf *metricFamilyPdata) getGroups() []*metricGroupPdata {
+	groups := make([]*metricGroupPdata, len(mf.groupOrders))
+	for k, v := range mf.groupOrders {
+		groups[v] = mf.groups[k]
+	}
+	return groups
+}
+
+func (mf *metricFamilyPdata) ToMetricPdata(metrics *pdata.MetricSlice) (int, int) {
+	metric := pdata.NewMetric()
+	pointCount := 0
+
+	switch mf.mtype {
+	case pdata.MetricDataTypeHistogram:
+		histogram := metric.Histogram()
+		hdpL := histogram.DataPoints()
+		for _, mg := range mf.getGroups() {
+			if !mg.toDistributionPoint(mf.labelKeysOrdered, &hdpL) {
+				mf.droppedTimeseries++
+			}
+		}
+		pointCount = hdpL.Len()
+
+	case pdata.MetricDataTypeSummary:
+		summary := metric.Summary()
+		sdpL := summary.DataPoints()
+		for _, mg := range mf.getGroups() {
+			if !mg.toSummaryPoint(mf.labelKeysOrdered, &sdpL) {
+				mf.droppedTimeseries++
+			}
+		}
+		pointCount = sdpL.Len()
+
+	default:
```

Comment:
This is missing a case for counters, but that seems to be covered in https://github.com/open-telemetry/opentelemetry-collector/pull/3741.

Reply:
Indeed, I added that case in the other PR but I've brought it in, please take a look again. Thank you @Aneurysm9!